<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
  <ros2_control name="AutonomousRoverSystem" type="system">
    <hardware>   
      <plugin>autonomous_rover/AutonomousRoverHardware</plugin>

      <!--Declare Motor Configurations-->
      <!-- pinForward,pinReverse,pinEnable,pwm_frequency-->
      <param name="left_motor">16,20,12,500</param>
      <param name="right_motor">6,5,13,500</param>
  
      <!--Declare Encoder Configurations-->
      <!-- pinA,pinB,cpr,wheel_circumference-->
      <param name="left_front_encoder">17,4,1920,0.205</param>
      <param name="left_back_encoder">22,27,1920,0.205</param>
      <param name="right_front_encoder">23,24,1920,0.205</param>
      <param name="right_back_encoder">25,8,1920,0.205</param>
   
      <!--Declare PID Controller Configurations-->
      <!-- enable,kp,ki,kd,limit-->
      <param name="left_controller">0,8.0,6.5,0.005,0.22</param>
      <param name="right_controller">0,8.5,6.5,0.005,0.22</param>

      <param name="min_velocity">0.3</param>
      <param name="max_velocity">1.0</param>

    </hardware>

    <!-- Wheel Joints -->
    <joint name="front_left_wheel_joint">
      <command_interface name="velocity">
        <param name="min">-5.0</param>
        <param name="max">5.0</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    
    <joint name="back_left_wheel_joint">
      <command_interface name="velocity">
        <param name="min">-5.0</param>
        <param name="max">5.0</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

    <joint name="front_right_wheel_joint">
      <command_interface name="velocity">
        <param name="min">-5.0</param>
        <param name="max">5.0</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

    <joint name="back_right_wheel_joint">
      <command_interface name="velocity">
        <param name="min">-5.0</param>
        <param name="max">5.0</param>
      </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

  </ros2_control> 

</robot>
